### PR TITLE
Update the pong tutorial + changelog for SpriteRenderData

### PR DESCRIPTION
**Pull request by Moxinilian**
This PR updates the pong tutorial in the book to have it reflect what is in the repo, after the addition of the `SpriteRenderData` utilities. It also officially adds the utilities to the changelog.

**Could the person who made the example screenshots update them? I have no idea how to make them clean and good looking like that.**

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/805)

### CODE REVIEW

File: book/src/pong_tutorial/pong_tutorial_02.md [hunk 1]
```diff
@@ -39,20 +39,90 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
...
+        ))
+        .build();
+}
+```
+
+We create an entity that will carry our camera,
+that we the create from an orthographic projection of the size of our
+arena (as we want it to cover it all). Ignore the `GlobalTransform` for now,
+we'll deal with it in more details later on.
+Note that we flipped the direction of the verical dimension by specifying
```
Comment (Xaeroxe) [inline book/src/pong_tutorial/pong_tutorial_02.md]: "vertical"

File: book/src/pong_tutorial/pong_tutorial_02.md [hunk 2]
```diff
@@ -39,20 +39,90 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
...
+        )))
+        .with(GlobalTransform(
+            Matrix4::from_translation(Vector3::new(0.0, 0.0, 1.0)).into()
+        ))
+        .build();
+}
+```
+
+We create an entity that will carry our camera,
+that we the create from an orthographic projection of the size of our
```
Comment (Xaeroxe) [inline book/src/pong_tutorial/pong_tutorial_02.md]: Grammar flow here is a bit odd, perhaps omit "that we the create"

File: book/src/pong_tutorial/pong_tutorial_02.md [hunk 3]
```diff
@@ -294,85 +314,90 @@ world
...
+Okay! We've defined our `Paddle` component, and created two entities which have
+a `Paddle` component, a `GlobalTransform` component and a sprite. When our game
+starts, we'll add the left and right paddles to the world, along with a camera.
 
-Now let's add our initialise functions to the `on_start` function in `impl State
-for Pong`.
+Before we continue, one last note.
+Components do not have to be registered manually to be used, however you need to have
+something that uses them to have them be registered automatically.
+As nothing uses our `Paddle` component yet, we will register it manually before we intialise
```
Comment (Xaeroxe) [inline book/src/pong_tutorial/pong_tutorial_02.md]: "initialize"

File: examples/pong_tutorial_01/main.rs [hunk 4]
```diff
@@ -27,9 +27,13 @@ fn main() -> amethyst::Result<()> {
         env!("CARGO_MANIFEST_DIR")
     );
 
+    // This line is not mentionned in the pong tutorial as it is specific to the context
```
Comment (Xaeroxe) [inline examples/pong_tutorial_01/main.rs]: "mentioned"

File: examples/pong_tutorial_02/main.rs [hunk 5]
```diff
@@ -14,10 +14,14 @@ fn main() -> amethyst::Result<()> {
         env!("CARGO_MANIFEST_DIR")
     );
 
+    // This line is not mentionned in the pong tutorial as it is specific to the context
```
Comment (Xaeroxe) [inline examples/pong_tutorial_02/main.rs]: "mentioned"

File: examples/pong_tutorial_03/main.rs [hunk 6]
```diff
@@ -21,14 +21,18 @@ fn main() -> amethyst::Result<()> {
         env!("CARGO_MANIFEST_DIR")
     );
 
+    // This line is not mentionned in the pong tutorial as it is specific to the context
```
Comment (Xaeroxe) [inline examples/pong_tutorial_03/main.rs]: "mentioned"

**Comment by Xaeroxe** [comment]
#803 may have a conflict with this, so be sure to rebase once it's merged in an hour or so.

**Comment by Xaeroxe** [comment]
@Moxinilian 

> Could the person who made the example screenshots update them? I have no idea how to make them clean and good looking like that.

Alt + Print Screen will probably help, then on Windows all you need to do is paste into something that can accept an image, such as the "Paint" program.

File: book/src/pong_tutorial/pong_tutorial_02.md (hunk 3 above)
Comment (Moxinilian) [reply book/src/pong_tutorial/pong_tutorial_02.md]: There is another instance of "initialise" that I did not write tho lol
I'll fix it too.

Comment (Moxinilian) [reply book/src/pong_tutorial/pong_tutorial_02.md]: I think that's american vs british here right?

Comment (Moxinilian) [reply book/src/pong_tutorial/pong_tutorial_02.md]: Ah I'm stupid nvm

Comment (Xaeroxe) [reply book/src/pong_tutorial/pong_tutorial_02.md]: The real reason this caught my attention was the missing second "i" so remember to get that too.

**Comment by Moxinilian** [comment]
Yeah I know, but the way they are cut is very clean. It has shading but no background. Plus it was made on macOS.

File: examples/pong_tutorial_01/main.rs (hunk 4 above)
Comment (Xaeroxe) [reply examples/pong_tutorial_01/main.rs]: @Moxinilian latest commit missed this one :P

Comment (Moxinilian) [reply examples/pong_tutorial_01/main.rs]: damn it

**Comment by Moxinilian** [comment]
Squashing is so easy with branches

File: book/src/pong_tutorial/pong_tutorial_02.md [hunk 7]
```diff
@@ -39,20 +39,89 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
...
+use amethyst::renderer::{
+    Camera, Event, PngFormat, Projection, Sprite, Texture, TextureHandle,
+    VirtualKeyCode, WithSpriteRender,
+};
 ```
 
+## Get around the World
+
+First, in `pong.rs`, let's add a new method to our State implementation: `on_start`.
+This method is called, as you probably guessed, on the start of the State.
```
Comment (jojolepro) [inline book/src/pong_tutorial/pong_tutorial_02.md]: on the start of the State. -> when the State starts

(the important part is the "when")

File: book/src/pong_tutorial/pong_tutorial_02.md [hunk 8]
```diff
@@ -39,20 +39,89 @@ stops complaining about unused imports. In `pong.rs` we'll need these use
...
+
+We create an entity that will carry our camera, with an orthographic projection
+of the size of our arena (as we want it to cover it all). Ignore the
+`GlobalTransform` for now, we'll deal with it in more details later on.
+Note that we flipped the direction of the vertical dimension by specifying
+`ARENA_HEIGHT` as the bottom. This is because by default bottom means
+the upper part of the screen, so here it will feel more natural to work with.
+
+> Orthographic projections are a type of 3D visualization on 2D screens
+> that keeps the size ratio of the 2D images displayed intact, very useful in
```
Comment (jojolepro) [inline book/src/pong_tutorial/pong_tutorial_02.md]: displayed intact, very useful  -> displayed intact. They are very useful

File: book/src/pong_tutorial/pong_tutorial_02.md [hunk 9]
```diff
@@ -132,28 +183,29 @@ to a parent if one exists, while `GlobalTransform` is, well, global.
...
 ```rust,ignore
-const PADDLE_HEIGHT: f32 = 0.30;
-const PADDLE_WIDTH: f32 = 0.05;
-const PADDLE_COLOUR: [f32; 4] = [0.0, 0.0, 1.0, 1.0];
+const PADDLE_HEIGHT: f32 = 16.0;
+const PADDLE_WIDTH: f32 = 4.0;
 ```
 
 Okay, let's make some entities! We'll define an `initialise_paddles` function
 which will create left and right paddle entities and attach `Transform`
```
Comment (jojolepro) [inline book/src/pong_tutorial/pong_tutorial_02.md]: attach `Transform` components to them -> attach a `Transform` component to each

**Comment by Moxinilian** [comment]
I also moved the spritesheet out of `assets/img` into `assets/texture`, I had misjudged the content of `img`.

**Comment by jojolepro** [comment]
Fix merge issues and I'll start the merge for you ;)

**Comment by Xaeroxe** [comment]
@Moxinilian hey can you rebase this so we can get it merged?

**Comment by Moxinilian** [comment]
Sure, as soon as I get back home (in less than 24 hours). 


12. Jul 2018 16:40 by notifications@github.com <mailto:notifications@github.com>:

**Comment by Moxinilian** [comment]
Okay I rebased everything and fixed a small mistake in tutorial 2 (regarding the explanation of the orthographic projection).
As soon as CI is done, it's ready for merging!

**Comment by Xaeroxe** [comment]
Thanks @Moxinilian !

bors r+

**Comment by jojolepro** [comment]
bors retry

**Pull request merged**